twingly_search/client.py

```diff
@@ -52,34 +62,41 @@ def query(self):
         """
         return Query(self)
 
-    def execute_query(self, query):
+    def execute_query(self, q):
```

Comment:
Would checking the type of the argument work?

Something like the following (it's in Ruby but you get the point :)):

```ruby
def execute_query(query)
  query_string = query.is_a?(Query) ? query.build_query_string : query
  response_body = self._get_response(query_string).content
  # ...
```